managers/BaseManager.js

```diff
@@ -7,54 +7,134 @@ const config = require('../common/config');
     const patchBody = _.cloneDeep(changeObjectBody);
     let currentAnnotations = patchBody.metadata.annotations;
     let patchAnnotations = currentAnnotations ? currentAnnotations : {};
     patchAnnotations.lockedByManager = config.broker_ip;
     patchBody.metadata.annotations = patchAnnotations;
-    return eventmesh.apiServerClient.updateResource(CONST.APISERVER.RESOURCE_GROUPS.BACKUP, CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP, changedOptions.guid, patchBody)
-      .tap((resource) => logger.info(`Successfully acquired processing lock for the backup request for backup guid: ${changedOptions.guid}\n` +
+    const resourceDetails = eventmesh.apiServerClient.parseResourceDetailsFromSelfLink(changeObjectBody.metadata.selfLink);
+    return eventmesh.apiServerClient.updateResource(resourceDetails.resourceGroup, resourceDetails.resourceType, changeObjectBody.metadata.name, patchBody)
+      .tap((resource) => logger.info(`Successfully acquired processing lock for request with options: ${JSON.stringify(changedOptions)}\n` +
         `Updated resource with annotations is: `, resource));
   }
 
   /**
    * @description Sets lockedByManager annotation to empty string
    */
 
-  static releaseProcessingLock(changeObjectBody) {
+  _releaseProcessingLock(changeObjectBody) {
     const changedOptions = JSON.parse(changeObjectBody.spec.options);
-    logger.info('Trying to release processing lock for the backup request for backup guid: ', changedOptions.guid);
+    logger.info('Trying to release processing lock for request with options: ', changedOptions);
     const patchBody = {
       metadata: {
         annotations: {
           lockedByManager: ''
         }
       }
     };
-    return eventmesh.apiServerClient.updateResource(CONST.APISERVER.RESOURCE_GROUPS.BACKUP, CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP, changedOptions.guid, patchBody)
-      .tap((resource) => logger.info(`Successfully released processing lock for the backup request for backup guid: ${changedOptions.guid}\n` +
+    const resourceDetails = eventmesh.apiServerClient.parseResourceDetailsFromSelfLink(changeObjectBody.metadata.selfLink);
+    return eventmesh.apiServerClient.updateResource(resourceDetails.resourceGroup, resourceDetails.resourceType, changeObjectBody.metadata.name, patchBody)
+      .tap((resource) => logger.info(`Successfully released processing lock for the request with options: ${JSON.stringify(changedOptions)}\n` +
         `Updated resource with annotations is: `, resource));
   }
 
+  _preProcessRequest(objectBody, processingLockStatus) {
+    const options = JSON.parse(objectBody.spec.options);
+    // Acquire processing lock so that in HA scenerio, only one backup-manager process processes the request
+    return Promise.try(() => {
+      if (!objectBody.metadata.annotations || objectBody.metadata.annotations.lockedByManager === '') {
+        return BaseManager.prototype._acquireProcessingLock(objectBody)
```

Comment:
I understand the need to use the `prototype` directly here since you do not have an object to work with but can this be done in a better way?